go-controller/pkg/ovn/controller/services/lb_config.go

```diff
@@ -832,3 +863,104 @@ func joinHostsPort(ips []string, port int32) []Addr {
 	}
 	return out
 }
+
+func getServicePortKey(protocol v1.Protocol, name string) string {
+	return fmt.Sprintf("%s/%s", protocol, name)
+}
+
+// GetEndpointsForService takes a service, all its slices and the list of nodes in the OVN zone
+// and returns two maps that hold all the endpoint addresses for the service:
+// one for cluster-wide endpoints, one for per-node endpoints.
+// The node list helps to keep the resulting map small, since we're only interested in local endpoints.
+func getEndpointsForService(slices []*discovery.EndpointSlice, service *v1.Service, nodes sets.Set[string]) (map[string]lbEndpoints, map[string]map[string]lbEndpoints) {
+	// classify endpoints
+	ports := map[string]int32{}
+	portToEndpoints := map[string][]discovery.Endpoint{}
+	portToNodeToEndpoints := map[string]map[string][]discovery.Endpoint{}
+
+	for _, port := range service.Spec.Ports {
+		name := getServicePortKey(port.Protocol, port.Name)
+		ports[name] = 0
+	}
+
+	for _, slice := range slices {
+
+		if !(slice.AddressType == discovery.AddressTypeIPv4 || slice.AddressType == discovery.AddressTypeIPv6) {
```

Comment:
Yeah, I can also just check against FQDN. I was checking against known values (ipv4, ipv6) because I'm a bit paranoid about future implementation of `AddressType`, but let's go for the simpler solution :)